.github/workflows/build-and-deploy.yml

```diff
@@ -80,6 +85,100 @@ jobs:
           SLACK_TITLE: 'Failure to Build API '
           SLACK_WEBHOOK: "${{ secrets.SLACK_WEBHOOK }}"
 
+
+  deploy_review_app:
+    name: Deploy to review environment
+    needs: build
+    if: ${{ (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'deploy')) || (github.event_name == 'workflow_dispatch' && inputs.environment == 'review' && inputs.pr-number != '') }}
+    runs-on: ubuntu-latest
+    concurrency: deploy_review_${{ github.event.pull_request.number || inputs.pr-number }}
+    permissions:
+      pull-requests: write
+      id-token: write
+    environment:
+      name: review
+
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: set-up-environment
+        uses: DFE-Digital/github-actions/set-up-environment@master
+
+      - uses: Azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Set PR Number
+        id: pr
+        run: |
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            echo "number=${{ inputs.pr-number }}" >> $GITHUB_OUTPUT
+          else
+            echo "number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+          fi
+
+      - uses: DFE-Digital/github-actions/set-kubelogin-environment@master
+        with:
+          azure-client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          azure-tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          azure-subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Deploy to Review Environment
+        uses: ./.github/workflows/actions/deploy_v2
+        id: deploy
+        with:
+          environment: review
+          sha: ${{needs.build.outputs.image_tag_sha}}
+          pr-number: ${{ steps.pr.outputs.number }}
+          azure-client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          azure-tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          azure-subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Get Review App URL
+        id: review_app
+        run: |
+          REVIEW_APP_URL=$(terraform -chdir=terraform/aks output -raw app_url)
+          echo "app_hostname=${REVIEW_APP_URL}" >> $GITHUB_OUTPUT
+
+      - name: Wait for Review App to be ready
```

Comment:
we run a healthcheck in the deploy_v2/action so not sure why we're running another, plus I don't believe /health exists